tests/test_other.py

```diff
@@ -12011,18 +12011,19 @@ def test_stack_switching_size(self):
     ''')
     expected = 'nap time\ni am awake\n'
 
-    shared_args = ['-Os', '-sEXIT_RUNTIME']
+    shared_args = ['-Os', '-sEXIT_RUNTIME', '-sENVIRONMENT=shell']
     self.run_process([EMXX, 'main.cpp', '-sASYNCIFY'] + shared_args)
     self.assertContained(expected, self.run_js('a.out.js'))
     asyncify_size = os.path.getsize('a.out.wasm')
 
-    self.run_process([EMXX, 'main.cpp', '-sASYNCIFY=2', '-sENVIRONMENT=shell'] + shared_args)
-    # run in v8 with stack switching and other relevant features (like reference
-    # types for the return value of externref)
-    v8 = config.V8_ENGINE + [
-      '--wasm-staging',
-      '--experimental-wasm-stack-switching'
-    ]
+    self.run_process([EMXX, 'main.cpp', '-sASYNCIFY=2'] + shared_args)
+
+    # enable stack switching and other relevant features (like reference types
+    # for the return value of externref)
+    self.v8_args.append('--wasm-staging')
+    self.v8_args.append('--experimental-wasm-stack-switching')
+    v8 = config.V8_ENGINE + self.v8_args
```

Comment:
You don't need this line.  `self.run_js` takes care of it and honors the `@require_v8`	above.

I'm trying to remove `engine=xx` from the tests and use the `requires_v8` or `self.require_v8()` consistently instead.

Reply:
Oh wait.. I'm getting confused with `js_engines` argument to the higher level functions... sorry about that.